### DISCUSSION TITLE
error message when running the input file

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#20406] [Q&A Modules: Phase field] [answered]
Hi everyone,

I got an error message when running my input file in supercomputing center. The following screenshot shows the error:
![image](https://user-images.githubusercontent.com/63679626/155624945-26fc9dc2-53aa-41fe-adf4-c16fb37beb51.png)
Why is this? My simulation stops and would not run.
Does anyone have any suggestions about this?

Thank you very much.
Best regards,
Haiying

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is this for MOOSE or a MOOSE derived app?
Please follow those instructions to get a backtrace for the error
https://mooseframework.inl.gov/application_development/debugging.html

Guillaume

**Reply by aprilnovak** [reply]
Is there more to the error message that may be explaining the issue? When MOOSE throws an error, typically the actual user-friendly error message is printed _before_ the MPI Abort, like this:

```*** ERROR ***
A 'PeripheralRingMeshGenerator' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
```

Here, the actual error message is telling you that `A 'PeripheralRingMeshGenerator' is not a registered object.`, which you may not notice if you just look at the very end of the error message that talks about MPI. Some new users I've helped have missed the actual error message like this.

Is there other info in your error message? If not, then @GiudGiud's suggestion of a backtrace is the best thing to do.
